x-pack/test_serverless/README.md

```diff
@@ -251,8 +256,9 @@ await samlAuth.setCustomRole({
 // Then, generate an API key with the newly defined privileges
 const roleAuthc = await samlAuth.createM2mApiKeyWithRoleScope('customRole');
 
-// Remember to invalidate the API key after use
+// Remember to invalidate the API key after use and delete the custom role
 await samlAuth.invalidateM2mApiKeyWithRoleScope(roleAuthc);
+await samlAuth.deleteCustomRole();
```

Comment:
So now the devs have to make two extra calls to finish?
Possible to make this a single call?

Reply:
I don't think we need to combine it.

There is nothing new in the workflow: API key and Kibana role are 2 different entities and we already handle "cleaning" in stateful separately. In theory API key is still valid after role deletion and can be used on its own. 

But now I realised it makes sense to delete cached cookie for custom role whenever `setCustomRole` or `deleteCustomRole` is called. Cookies are cached per role on the first call to speedup test execution, but for as soon as custom role is re-created we need to regenerate cookie as well. I will test it and address in the new PR

Reply:
small fix dbdcd59ef2276adb6554bda3e976e8b3e9d3a538

tl; dr; generated cookie remains valid after roles is re-created after deletion or its privileges were updated. I don't think we need a logic to drop it from cache at the moment